File: src/main/java/ru/melonhell/bulletphysics/utils/math/Convert.java
package ru.melonhell.bulletphysics.utils.math;

import com.jme3.math.Vector3f;
import org.bukkit.util.Vector;
import ru.melonhell.bulletphysics.nms.wrappers.BlockPosWrapper;

public class Convert {
    public static Vector3f toBullet(BlockPosWrapper blockPos) {
        return new Vector3f(blockPos.getX() + 0.5f, blockPos.getY() + 0.5f, blockPos.getZ() + 0.5f);
    }

    public static Vector toMinecraft(Vector3f vector3f) {
        return new Vector(vector3f.x, vector3f.y, vector3f.z);
    }

    public static Vector3f toBullet(Vector vector3f) {
        return new Vector3f((float) vector3f.getX(), (float) vector3f.getY(), (float) vector3f.getZ());
    }
}

File: src/main/java/ru/melonhell/bulletphysics/utils/math/Convert.kt
package ru.melonhell.bulletphysics.utils.math

import com.jme3.math.Vector3f
import org.bukkit.util.Vector
import ru.melonhell.bulletphysics.nms.wrappers.BlockPos

object Convert {
    fun toBullet(blockPos: BlockPos): Vector3f {
        return Vector3f(blockPos.x + 0.5f, blockPos.y + 0.5f, blockPos.z + 0.5f)
    }

    fun toMinecraft(vector3f: Vector3f): Vector {
        return Vector(vector3f.x, vector3f.y, vector3f.z)
    }

    @JvmStatic
    fun toBullet(vector3f: Vector): Vector3f {
        return Vector3f(vector3f.x.toFloat(), vector3f.y.toFloat(), vector3f.z.toFloat())
    }
}